src/TransferList/TransferList.test.js

```diff
@@ -0,0 +1,98 @@
+import { render, screen, within } from "@testing-library/react";
+import React from "react";
+import TransferList from ".";
+import userEvent from "@testing-library/user-event";
+
+/**
+ * Test wrapper for TransferList
+ * Provides state for the selectedItems
+ */
+const SelectedItemsWithState = ({
+  items: itemsIn = ["Apples", "Pears", "Oranges"],
+  onChange,
+  selectedItem: selectedItemsIn = ["Apples"],
+  ...rest
+}) => {
+  const [selectedItems, setSelectedItems] = React.useState(selectedItemsIn);
+  const handleChange = selectedItem => {
+    setSelectedItems(selectedItems);
+    onChange && onChange(selectedItems);
+  };
+  return (
+    <TransferList
+      {...rest}
+      items={itemsIn}
+      onChange={handleChange}
+      selectedItems={selectedItems}
+    />
+  );
+};
+
+describe("TransferList", () => {
+  test("Search bar updates value", async () => {
+    const user = userEvent.setup();
+    const { container } = render(<SelectedItemsWithState />);
+
+    const inputBase = container.querySelector(".MuiInputBase-input");
+    await user.type(inputBase, "pears");
+    expect(inputBase.value).toBe("pears");
```

Comment:
This isn't really checking the correct thing. All this test is doing is typing in search box and then checking that value in search box is correct, which of course it will be because user.type deals with that. We don't need to test user.type functionality. What we should be testing is that when search text is typed then the list on the left is filtered and shows the correct items.